### DISCUSSION TITLE
Docs on IOC in Adonis 5

### DISCUSSION THREAD

**Discussion by tslocke** [adonisjs/core#3013] [Help]
Hi all,

Searches for docs on the providers and IOC in Adonis bring up lots of the legacy docs pages. It seems there was whole section in the official docs on service providers.

A big change in Adonis 5 seems to be that `use()` is no longer needed, so I'm assuming these docs are out of date.

The only page I could find about the new stuff is

https://dev.to/shemsiu/ioc-container-and-dependency-injection-in-adonis-v5-og2

Is there anything else around?

My specific need is to write a provider for the arangodb client. It's not that big a deal, and I already have it working as a regular JS module, but it would be good to do this properly.

Thanks

Tom

**Comment by McSneaky** [comment]
Hello!

There's no doc for it currently indeed (hopefully it's coming soon)

I guess ArongoDB client would be somewhat similar to MongoDB client provider
Wrote a small guide about MongoDB provider in here: https://mcsneaky.ap3k.pro/posts/adonis-v5-provider/ 
Might come handy
